lib/restforce.rb

```diff
@@ -42,7 +42,7 @@ module Tooling
   Error               = Class.new(StandardError)
   ServerError         = Class.new(Error)
   AuthenticationError = Class.new(Error)
-  UnauthorizedError   = Class.new(Error)
+  UnauthorizedError   = Class.new(Faraday::Error::ClientError)
```

Comment:
Why do you think this should inherit from a Faraday error when the others don't?

Reply:
I think the others should probably derive from ClientError as well, but this is the only Error type referenced in `lib/restforce/middleware/raise_error.rb` that does not have support for the response attribute. I'm honestly not sure when any of the other errors would really be used since they are not referenced in the `raise_error` middleware.

I'm specifically looking at the following:
https://github.com/restforce/restforce/blob/c79827f5ceb37fe32f21b21652c9a339bf733abb/lib/restforce/middleware/raise_error.rb#L19-L21

Reply:
Makes sense!